tests/protocol/test_metadata_server_migration_util.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+
+import os
+import tempfile
+import unittest
+
+import azurelinuxagent.common.osutil.default as osutil
+import azurelinuxagent.common.protocol.metadata_server_migration_util as migration_util
+
+from azurelinuxagent.common.protocol.metadata_server_migration_util import _LEGACY_METADATA_SERVER_TRANSPORT_PRV_FILE_NAME, \
+                                                                           _LEGACY_METADATA_SERVER_TRANSPORT_CERT_FILE_NAME, \
+                                                                           _LEGACY_METADATA_SERVER_P7B_FILE_NAME, \
+                                                                           _KNOWN_METADATASERVER_IP
+from azurelinuxagent.common.utils.restutil import KNOWN_WIRESERVER_IP
+from tests.tools import AgentTestCase, patch, call
+
+class TestMetadataServerMigrationUtil(AgentTestCase):
+    def test_ensure_file_removed(self):
```

Comment:
we try to avoid testing private methods
_ensure_file_removed should be covered by the other tests

Reply:
similar for tests below